### PR TITLE
Add .npmignore

### LINKED ISSUE #232: Add `.npmignore`
You are shipping various files that are redundant for the end user such as:

```
node_modules/generic-pool/.eslintignore
node_modules/generic-pool/.eslintrc.js
node_modules/generic-pool/.travis.yml
node_modules/generic-pool/CHANGELOG.md
node_modules/generic-pool/Makefile
node_modules/generic-pool/tsconfig.json
node_modules/generic-pool/test/
node_modules/generic-pool/test/doubly-linked-list-iterator-test.js
node_modules/generic-pool/test/doubly-linked-list-test.js
node_modules/generic-pool/test/generic-pool-acquiretimeout-test.js
node_modules/generic-pool/test/generic-pool-test.js
node_modules/generic-pool/test/resource-request-test.js
node_modules/generic-pool/test/utils.js
```

### PR DESCRIPTION
**Pull request by t3hmrman**
Resolves #232

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+node_modules/generic-pool/.eslintignore
```
Comment (SimonSchick) [inline .npmignore]: Aren't the paths here wrong, shouldn't these be relative to the project root?

Comment (sandfox) [reply .npmignore]: errm.., yes - those look wrong now you mention it! I'm half asleep this morning!

Comment (sandfox) [reply .npmignore]: this is reverted now, it was never published, but if you're using this module via a github ref you'll want to update

Comment (SimonSchick) [reply .npmignore]: lol

**Pull request merged**